endpoints/cookie_sync.go

```diff
@@ -427,6 +427,20 @@ func (c *cookieSyncEndpoint) handleResponse(w http.ResponseWriter, tf usersync.S
 		})
 	}
 
+	if debug {
+		biddersSeen := make(map[string]struct{})
+		var debugMessages []string
+		for _, bidderEval := range biddersEvaluted {
+			if bidderEval.Status == usersync.StatusDuplicate && biddersSeen[bidderEval.Bidder] == struct{}{} {
+				debugMessages = append(debugMessages, getDebugMessage(bidderEval.Status)+" synced as "+bidderEval.SyncerKey)
+			} else if bidderEval.Status != usersync.StatusDuplicate {
+				debugMessages = append(debugMessages, getDebugMessage(bidderEval.Status))
+			}
+			biddersSeen[bidderEval.Bidder] = struct{}{}
+		}
+		response.Debug = debugMessages
+	}
```

Comment:
I tested it with activities where only  "pubmatic" and "appnexus" are allowed. 
Request to cookie_sync: 
```json
{
    "bidders": ["appnexus", "pubmatic", "adf", "synacormedia"],
    "debug":true
}
```
Expected: 2 bidders ("adf", "synacormedia") are rejected by privacy and 2 bidders are allowed.
Here is debug part of response:
```json
"debug": [
        "",
        "Rejected by privacy",
        "Rejected by privacy",
        ""
    ]
```
Can you please add bidder name to message, f.i  "Rejected by privacy: synacormedia" and remove empty strings from debug messages. 
In activity requirements it states that message should be "Bidder sync blocked for privacy reasons". Not sure what our message should be, I feel like having bidder name in it is reasonable. 

Also, about debug flag in request. Here it is a field in json body: `"debug": true`
Is it confirmed it should be like this? 
I'm asking because we handle debug flag differently in different endpoints. F.i. in `/action`  it is `"test": 1`

Reply:
@AlexBVolcy Please match the response format established with PBS-Java.

Reply:
I agree here. I also want to point out that it seems that some of our syncer error messages are a bit different from PBS-Java. I think that's ok. @SyntaxNode, @VeronikaSolovei9 thoughts? What's most important is that the request and response formats are the same.

Reply:
We should match the debug messages too. Unless they reference the "cookie family", which is called the "syncer key" in Go.

Reply:
So here are the debug messages that PBS-Java uses
```
builder = switch (reason) {
    case INVALID_BIDDER -> builder.conditionalError(requested, "Unsupported bidder");
    case DISABLED_BIDDER -> builder.conditionalError(requested, "Disabled bidder");
    case REJECTED_BY_TCF -> builder.conditionalError(requested || coopSync, "Rejected by TCF");
    case REJECTED_BY_CCPA -> builder.conditionalError(requested || coopSync, "Rejected by CCPA");
    case DISALLOWED_ACTIVITY -> builder.conditionalError(requested || coopSync, "Disallowed activity");
    case UNCONFIGURED_USERSYNC -> builder.conditionalError(requested, "No sync config");
    case DISABLED_USERSYNC -> builder.conditionalError(requested || coopSync, "Sync disabled by config");
    case REJECTED_BY_FILTER -> builder.conditionalError(requested || coopSync, "Rejected by request filter");
    case ALREADY_IN_SYNC -> builder.conditionalError(requested, "Already in sync");
};
```
I tried to match debug messages the best I could, but there are some statues in my PR that don't overlap with the PBS-Java statuses (i.e. `StatusTypeNotSupported`), so in those cases I came up with my own debug message. Is this alright?

Reply:
And yes @VeronikaSolovei9, the debug flag in the request should be like this `debug: true`

Reply:
Thank you for addressing comments, just checked it locally and it works as expected. 
Question: for bidders restricted by activity here is an expected message:
`case DISALLOWED_ACTIVITY -> builder.conditionalError(requested || coopSync, "Disallowed activity");`
In this PR we have `"Rejected by privacy: synacormedia"` (I know I asked about the bidder name).  
Should it be? `Disallowed activity`?